src/NUnitFramework/tests/Attributes/TestCaseSourceTests.cs

```diff
@@ -290,6 +290,21 @@ public void HandlesExceptionInTestCaseSource_GuiDisplay(string lhs, string rhs)
             Assert.AreEqual(lhs, rhs);
         }
 
+
+        private static IEnumerable<TestCaseData> ZeroTestCasesSource() => Enumerable.Empty<TestCaseData>();
+
+        [TestCaseSource(nameof(ZeroTestCasesSource))]
+        public void Test_with_zero_test_cases_should_not_be_run()
+        {
+            Assert.Fail("Test should not be run.");
+        }
+
+        [TestCaseSource(nameof(ZeroTestCasesSource))]
+        public void Test_with_zero_test_cases_should_pass_without_requiring_arguments(int requiredParameter)
+        {
+        }
+
+
```

Comment:
Well, #1933 is an unreviewed new issue, so we don't actually have a goal yet. :smile: